FitGoal/FitGoal/Singup&Singin/CustomUIViews/AuthenticationFormStack.swift

```diff
@@ -0,0 +1,177 @@
+//
+//  SingUpForm.swift
+//  FitGoal
+//
+//  Created by Eliany Barrera on 20/3/20.
+//  Copyright © 2020 Eliany Barrera. All rights reserved.
+//
+
+import UIKit
+
+class AuthenticationFormStack: UIStackView, UITextFieldDelegate {
+    
+    convenience init(type: AuthenticationType) {
+        self.init(frame: .zero)
+        switch type {
+        case .login:
+            let emailAddress = TextFieldType.emailAdress.getCustomTextField()
+            let password = TextFieldType.password.getCustomTextField()
+            
+            let textFields = [emailAddress, password]
+            addArrengedSubviews(textFields)
+            setTextFieldHeightConstraints(for: textFields)
+            setDelegate(for: textFields)
+            configureReturnKeyType(for: textFields)
+        case .signUp:
+            let name = TextFieldType.userName.getCustomTextField()
+            let emailAddress = TextFieldType.emailAdress.getCustomTextField()
+            let password = TextFieldType.password.getCustomTextField()
+            let confirmPassword = TextFieldType.confirmPassword.getCustomTextField()
+            
+            let textFields = [name, emailAddress, password, confirmPassword]
+            addArrengedSubviews(textFields)
+            setTextFieldHeightConstraints(for: textFields)
+            setDelegate(for: textFields)
+            configureReturnKeyType(for: textFields)
+        case .none:
```

Comment:
Yes! Was expecting that! I was going to give you an answer but I realised I included this case because I interpreted the design in the wrong way. Will try to improve this